storage/innobase/xtrabackup/doc/source/version-check.rst

```diff
@@ -4,63 +4,113 @@
+community.
 
-The feature allows compatible tools to query the Version Check database and
-print advice on any potential risks associated with any of the relevant
-software versions.
+
+The purpose of this document is to articulate the information that is
+collected, as well as to provide guidance on how to disable this functionality
+if desired.
 
 Usage
------
+=====
 
 *Version Check* was implemented in |pt| 2.1.4, and was enabled by default in
-version 2.2.1. Currently it is supported by most tools in |pt|, |pxb|, and
-|pmm|. In |Percona XtraBackup|, this feature was introduced in version 2.1.5. By
-default, it is also enabled.
+version 2.2.1. Currently it is supported as a ``--[no]version-check`` option
+by `a number of tools in Percona Toolkit <https://www.percona.com/doc/percona-toolkit/LATEST/genindex.html>`_,
+|pxb|, and |pmm|.
 
-Being called with Version Check enabled, the tool that supports this feature
-connects to a dedicated Percona server via a secure HTTPS channel. It checks
-its own version to query the server for possible updates, and also checks
+When launched  with Version Check enabled, the tool that supports this feature
+connects to a Percona's *version check service* via a secure HTTPS channel. It
+compares the locally installed version for possible updates, and also checks
 versions of the following software:
 
-* operating system
-* |Percona XtraBackup|
-* |MySQL|
+* Operating System
+* Percona Monitoring and Management (PMM)
+* MySQL
 * Perl
-* |MySQL| driver for Perl (DBD::mysql)
-* |pt|
+* MySQL driver for Perl (DBD::mysql)
+* Percona Toolkit
+
+Then it checks for and warns about versions with known problems if they are
+identified as running in the environment.
+
+Each version check request is logged by the server. Stored information consists
+of the checked system unique ID followed by the software name and version.
+The ID is generated either at installation or when the |version-check| query is
+submitted for the first time.
+
+.. note::
 
-Then, it checks for and warns about versions with known problems.
+   Prior to version 3.0.7 of |pt|, the system ID was calculated as an MD5 hash
+   of a hostname, and starting from |pt| 3.0.7 it is generated as an MD5
+   hash of a random number. |pxb| continues to use hostname-based MD5 hash.
 
-To guide the development of future |version-check| requirements, each request is
-logged by the server. Stored information includes software version numbers and
-the unique ID of a checked system. The ID (an MD5 hash value) is generated
-either at installation or when the |version-check| query is done for the first
-time.
+As a result, the content of the sent query is as follows::
+
+  85624f3fb5d2af8816178ea1493ed41a;DBD::mysql;4.044
+  c2b6d625ef3409164cbf8af4985c48d3;MySQL;MySQL Community Server (GPL) 5.7.22-log
+  85624f3fb5d2af8816178ea1493ed41a;OS;Manjaro Linux
+  85624f3fb5d2af8816178ea1493ed41a;Percona::Toolkit;3.0.11-dev
```

Comment:
Does PXB also sending PT version?

Reply:
Sergei, yes, the version check script in the PXB repo reports toolkit version.   Sadly,  it is not configured to report PXB version.   There is a another jira ticket for this: PXB-1615

Hopefully we will get this resolved before the PXB 8.0.1 release